spring-integration-file/src/main/java/org/springframework/integration/file/remote/aop/RotatingServerAdvice.java

```diff
@@ -88,167 +84,28 @@ public boolean beforeReceive(MessageSource<?> source) {
 		return result;
 	}
 
-	/**
-	 * Implementations can reconfigure the message source before and/or after
-	 * a poll.
-	 */
-	public interface RotationPolicy {
```

Comment:
We have to revert this change. It is fully blown breaking change in the point version.
I'm OK with a decoupling in the current `5.2` (preparing some polishing PR), but I'm against any breaking changes for public API in the point release.

Does it make sense, @garyrussell , @dturanski ?

Thanks

Reply:
Technically you're right. But I think it would be ok if we default it. Then it won't break anything. I'd be surprised if anyone else has actually implemented this interface.

Reply:
We just can't guess from here. This interface is `public` in this place, therefore moving it somewhere else is a breaking change.
The first rule: don't break any public API in the point release!

Reply:
I see, because I moved it from the inner class.  Yes that is a breaking change.

Reply:
Good. So, @garyrussell , your turn now and then we will make a final decision!